### DISCUSSION TITLE
Increasing the number of photos users upload is better for generating better quality images?

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#2881] [Q&A] [answered]
Hello, community 😊
I 've tested this application for image generation with text prompt + image prompt 

As a text prompt "Wedding guest"
I tried to upload 1 ~ 4 images as image prompt and set to "face swap"

The result is amazing but some times I got several issues with output face's issues especially eyes, and their hands and legs.
I just need some advice how to resolve this issue and would like to know if increasing number of photos upload would work or not.

There is maximum 4 images which can be uploaded.

What if we can upload 6  or 10 images? If then can we expect the better result?

Looking forward to good response.
Thank you.

**Comment by ILikeToasters** [comment]
Often you will need to use inpainting to touch things up. AI is not perfect with generating things and some of it is just luck. If you want to utilize faceswap when inpainting you will need to check the box "Mixing Image Prompt and Inpaint" in the Control tab under the debug mode.
<img width="350" alt="Screenshot 2024-05-07 111042" src="https://github.com/lllyasviel/Fooocus/assets/4357657/70ec2d38-d10a-441c-8fda-4a58ed0a7601">

**Comment by mashb1t** [comment, marked as answer]
@oldhand7 you can test if increasing the amount of reference images works better.
This can be done by adjusting this line in the code: https://github.com/lllyasviel/Fooocus/blob/main/modules/flags.py#L92

**Reply by oldhand7** [reply]
Thank you for reply. I would like to know if Increasing the number of photos could impact server performance and there's a trade-off involved. And furthermore, I can't figure it out if it works better or not. So I need some advice how it works better for output generation.

Hope to hear from you.

**Reply by mashb1t** [reply]
Yes, it will impact server performance as Fooocus needs to (pre)process more IP adapters / ControlNets.
You can best tell if an image looks better than anyone else could as you're the one prompting and knowing what you wish for. Please consider using the same seed and then use >4 image prompts to check if the impact is significant enough for it to make a difference and how well the results may look like. 4 IP adapters is the recommended amount as it offers the best balance between usability and required processing performance.

**Reply by oldhand7** [reply]
Okay. Thanks. Maybe this is silly question but I am just excited about this application and would like to know the main idea of seed, IP adapters and their controlNets and so on. Furthermore, I would like to know all the angles of the project. I am a AI begineer.
Please let me know how to learn this and their architectures.
Thank you.

**Reply by mashb1t** [reply]
You can familiarize yourself with the basics here: https://stable-diffusion-art.com/beginners-guide/

If you have more specific questions about Fooocus, feel free to ask.
